crypto/err/err.c

```diff
@@ -664,6 +664,9 @@ ERR_STATE *ERR_get_state(void)
 {
     ERR_STATE *state = NULL;
 
+    if (!OPENSSL_init_crypto(OPENSSL_INIT_BASE_ONLY, NULL))
+        return NULL;
```

Comment:
Does it need to get called in drbg_lib.c before all RUN_ONCE()s too?

Reply:
I am not sure, maybe as another PR.
It would be good for instance, if OPENSSL_thread_stop()
was able to clean up that thread state.
I don't know if that this function does currently with the per thread DRBGs.

Reply:
There is some lack of internal documentation on how to properly use those thread local keys, and it took a while for me to make sure it all passed without leaking. What I currently remember:
We only want 1 function that sets a destructor for the local key to be called when the thread it stopped. This is because detection of thread stop works differently on Windows, and so we have 1 function that cleans up all thread local data. You need to modify ossl_init_thread_stop and ossl_init_thread_start to keep track that the threads has thread local data or not, and so needs to be cleaned up or not. And you need to call ossl_init_thread_start in your RUN_ONCE funtion.

Anyway, we have thread local data for error, async and rand. My guess is that they all need to call this new function.

Reply:
Async does not need to change.
The problem with the error module is that the init function creates a recursion,
So OPENSSL_INIT_BASE_ONLY is only needed here.
The other modules  just have to make sure they call OPENSSL_init_crypto
before they allocate their thread local keys.
And they should better use ossl_init_thread_start which rand doesn't do currently.